### ISSUE TITLE
Bug Version 2.76.0 azurerm_kubernetes_cluster_node_pool adds by default a os_sku  = "Ubuntu" when I deploy a windows node and it fails

### ISSUE THREAD

**Issue by jm01** [hashicorp/terraform-provider-azurerm#13306]
The new version 2.76.0 adds a  os_sku  = "Ubuntu" when I deploy a windows node. Thus, it fails

Error: creating/updating Managed Kubernetes Cluster Node Pool "uwinnd" (Resource Group "z0euw1dcmrsg002"): containerservice.AgentPoolsClient#CreateOrUpdate: Failure sending request: StatusCode=0 -- Original Error: Code="InvalidOSSKU" Message="OSSKU='Ubuntu' is invalid, details: Windows does not allow OSSKU selection"

The plan:
# azurerm_kubernetes_cluster_node_pool.nodes["1"] will be created
  + resource "azurerm_kubernetes_cluster_node_pool" "nodes" {
      + availability_zones    = [
          + "1",
          + "2",
        ]
      + enable_auto_scaling   = true
      + id                    = (known after apply)
      + kubelet_disk_type     = (known after apply)
      + kubernetes_cluster_id = "/subscriptions/54f6184f-874c-4082-b79a-269f5641cd1c/resourcegroups/z0euw1dcmrsg002/providers/Microsoft.ContainerService/managedClusters/caasplatform01dakspltfrm"
      + max_count             = 5
      + max_pods              = (known after apply)
      + min_count             = 2
      + mode                  = "User"
      + name                  = "uwinnd"
      + node_count            = 5
      + node_labels           = (known after apply)
      + orchestrator_version  = (known after apply)
      + os_disk_size_gb       = (known after apply)
      + os_disk_type          = "Managed"
      + os_sku                = "Ubuntu"
      + os_type               = "Windows"
      + priority              = "Regular"
      + spot_max_price        = -1
      + ultra_ssd_enabled     = false
      + vm_size               = "Standard_D4s_v3"
      + vnet_subnet_id        = "/subscriptions/54f6184f-874c-4082-b79a-269f5641cd1c/resourceGroups/z0euw1dcmrsg002/providers/Microsoft.Network/virtualNetworks/caasplatform01daksplatformvn/subnets/caasplatform01daksplatformkubesubnet"
    }

My Script

*****************************************************

locals {
  node_pools = [
     {
        name   = "ulnxnd"
        node_count          = var.aks_agent_count_linuxuser
        vm_size             = var.aks_agent_vm_size_linuxuser
        type                = "VirtualMachineScaleSets"
        availability_zones  = ["1", "2"]
        enable_auto_scaling = true
        min_count           = var.min_count_linuxuser
        max_count           = var.max_count_linuxuser
        mode                = "User"
        os_type             = "Linux"
     },
     {
        name                  = "uwinnd"
        node_count          = var.aks_agent_count_windowsuser
        vm_size             = var.aks_agent_vm_size_windowsuser
        type                = "VirtualMachineScaleSets"
        availability_zones  = ["1", "2"]
        enable_auto_scaling = true
        min_count           = var.min_count_windowsuser
        max_count           = var.max_count_windowsuser
        vnet_subnet_id      = azurerm_subnet.kubesubnet.id
        mode                = "User"
        os_type             = "Windows"
     }
  ]
}

resource "azurerm_kubernetes_cluster_node_pool" "nodes" {
 lifecycle {
    ignore_changes = [
      node_count
    ]
  }
  for_each              = {
    for index, node in local.node_pools:
    index => node }
  name                  = "${each.value.name}"
  kubernetes_cluster_id = azurerm_kubernetes_cluster.cluster.id
  node_count            = each.value.node_count
  vm_size               = each.value.vm_size
  availability_zones    = each.value.availability_zones
  enable_auto_scaling   = each.value.enable_auto_scaling
  min_count             = each.value.min_count
  max_count             = each.value.max_count
  vnet_subnet_id        = azurerm_subnet.kubesubnet.id
  mode                  = each.value.mode
  os_type               = each.value.os_type

  
}
**************************************************************************

**Comment by f35f0ef9d0e827dae86552d3899f78fc** [comment]
Linking to related feature-add: #13284

**Comment by smark88** [comment]
This will also force a cluster rebuild when set even tho nothing is changing. We default to Ubuntu and now its setting 
```
+ os_sku                       = "Ubuntu" # forces replacement
```
and wanting to force all our clusters to rebuild.

**Comment by BertelBB** [comment]
> This will also force a cluster rebuild when set even tho nothing is changing. We default to Ubuntu and now its setting
> 
> ```
> + os_sku                       = "Ubuntu" # forces replacement
> ```
> 
> and wanting to force all our clusters to rebuild.

Exactly! It is also marked as _Optional_ which it clearly is not

**Comment by aristosvo** [comment]
@katbyte this seems to be already fixed by #13321, which is released in 2.77.0

**Comment by stephybun** [comment]
As pointed out [in this comment](https://github.com/hashicorp/terraform-provider-azurerm/issues/13306#issuecomment-921770185) this issue has been resolved and released in v2.77.0.
